ouroboros-network/test/Test/AnchoredFragment.hs

```diff
@@ -162,37 +162,37 @@ prop_addBlock (TestAddBlock c b) =
 
 prop_rollback :: TestAnchoredFragmentAndPoint -> Bool
 prop_rollback (TestAnchoredFragmentAndPoint c p) =
-    case AF.rollback p c of
-      Nothing -> not (AF.withinFragmentBounds p c)
+    case AF.rollback (Point p) c of
+      Nothing -> not (AF.withinFragmentBounds (Point p) c)
       Just c' ->
         -- chain is a prefix of original
            AF.isPrefixOf c' c
         -- chain head point is the rollback point
-        && AF.headPoint c' == p
+        && AF.headPoint c' == Point p
 
 prop_rollback_head :: TestBlockAnchoredFragment -> Bool
 prop_rollback_head (TestBlockAnchoredFragment c) =
     AF.rollback (AF.headPoint c) c == Just c
 
 prop_successorBlock :: TestAnchoredFragmentAndPoint -> Property
 prop_successorBlock (TestAnchoredFragmentAndPoint c p) =
-  AF.withinFragmentBounds p c ==>
-  case AF.successorBlock p c of
-    Nothing -> AF.headPoint c === p
-    Just b  -> property $ AF.withinFragmentBounds (AF.blockPoint b) c
-          .&&. blockPrevHash b === pointHash p
+  AF.withinFragmentBounds (Point p) c ==>
+  case AF.successorBlock (Point p) c of
+    Nothing -> AF.headPoint c === Point p
+    Just b  -> property $ AF.withinFragmentBounds (Point (AF.blockPoint b)) c
+          .&&. blockPrevHash b === BlockHash (pointHash p)
```

Comment:
Lots and lots of BlockPoint -> Point conversion going on. I wonder if we couldn't pick some generators or utils so that we don't need quite so many.